### ISSUE TITLE
Error Starting New Game on Fresh Install

### ISSUE THREAD

**Issue by jrow3** [kestasjk/webDiplomacy#263]
# Bug report

Error when trying to start a New Game on a fresh install

Error triggered: file_put_contents(variants/Modern2/cache/territories.js): failed to open stream: No such file or directory

(error edited)

## Expected behaviour

New Game prompt

## Actual behaviour

Error triggered: file_put_contents(variants/Modern2/cache/territories.js): failed to open stream: No such file or directory

Error: "file_put_contents(variants/Modern2/cache/territories.js): failed to open stream: No such file or directory"
Raised: "/home/apog/djrow3.com/diplomacy/variants/install.php"
Line: "89"
userID = 6Variable dump: jsonFileLocation => variants/Modern2/cache/territories.js
mapID => 19
DB => Array: (
putqueries => 11
getqueries => 12
)
territories => Array: (
1 => Array: (

## Steps to reproduce

Continued error.

**Comment by TimothyJones** [comment]
Thanks for the report. I think webdiplomacy doesn't create the cache directories by default. 

Does `variants/Modern2/cache/` exist? If not, you should be able to work around this by creating it.

**Comment by TimothyJones** [comment]
We can fix this by creating the cache directories by default.

**Comment by jrow3** [comment]
I was able to bypass this error by copy and pasting over the Classic/cache folders to Modern2 and Empire4.  It seems the master does not include these files.

**Comment by TimothyJones** [comment]
Thanks! A pull request that contains that change would be welcome if you have the time to make one.

**Comment by jrow3** [comment]
Will do tomorrow

**Comment by TimothyJones** [comment]
I've added those files in commit https://github.com/kestasjk/webDiplomacy/commit/570a630a63946d90b7235397f16dadfd680ae746

**Comment by TimothyJones** [comment]
Yes, we should create the cache directories when they don't exist.

**Comment by Sleepcap** [comment]
These should be moved to the main-cache-directory (cache/variants/VARIANT_SHORTNAME for example)
Storing cached-data in each variant-subdirectory is not really good coding practice.
Maybe use the /lib/cache.php to handle these too...

**Comment by jmo1121109** [comment]
@Sleepcap if this is something you'd feel comfortable doing on vDip I'd happily put the time into porting it back over here. I just don't know how to do it myself.
